tests/app/test_jinja_filters.py

```diff
@@ -95,6 +96,28 @@ def test_format_date(self):
 
         self.assertEqual(format_value, '1 January 2017')
 
+    def test_format_conditional_date(self):
+        # Given
+        date = '03/11/2017'
+        date2 = '04/08/2018'
+        dates = ([date, date2])
+
+        # When
+        format_value = format_conditional_date(dates)
+
+        self.assertEqual(format_value, '11 March 2017')
+
+    def test_format_conditional_date2(self):
+        # Given
+        date = None
+        date2 = '10/12/17'
+        dates = ([date, date2])
+
+        # When
+        format_value = format_conditional_date(dates)
+
+        self.assertEqual(format_value, '12 October 2017')
+
```

Comment:
We should have more unit tests. Currently there are no tests for when dates passed to this method are Python Date objects, and there are no tests with just the first date argument set.